### PR TITLE
locks down ami_builder IAM user to a single VPC

### PR DESCRIPTION
**Pull request by jonauman**
We want to ensure that the IAM user can create/destroy instances
in a specific VDC only to protect other VDCs from accidental or
malicious shutdown of instances.  IAM permissions are quite complicated
and unintuitive. After much trial and error, we successfully followed
the instructions in this Amazon post: http://amzn.to/1ouXPo0

In essence:
1) create a role with no policy attached (VPCLockDown)
2) In the user policy, give the user access to:
  - assign the role to instances
  - control instances with the role
  - bring up instances in a specific VPC (only)
  - global permissions to describe things (to discover resources)

Whereas it's not really obvious why we need to create an instance role,
that is the key to making the user policy work.

### CODE REVIEW

**Comment by deanwilson** [comment]
Looks good. Merge away.

File: packer/ami.json [hunk 1]
```diff
@@ -4,10 +4,13 @@
...
       "type": "amazon-ebs",
       "ami_name": "govuk-trusty-1404-{{isotime \"2006-01-02T1504\"}}",
+      "iam_instance_profile": "VPCLockDown",
       "instance_type": "t2.micro",
       "region": "eu-west-1",
       "source_ami": "ami-b0c379c3",
-      "ssh_username": "ubuntu"
+      "ssh_username": "ubuntu",
+      "subnet_id": "subnet-179f5e73",
+      "vpc_id": "vpc-255a4440"
```
Comment (mattbostock) [inline packer/ami.json]: Can we avoid hard-cording these values?

File: projects/packer_ami_builder/resources/production/templates/ami_building_policy.tpl [hunk 2]
```diff
@@ -1,39 +1,107 @@
...
-          "StringEquals": {
-            "ec2:vpc": "arn:aws:ec2:${region}:${account_id}:vpc/${vpc_id}"
-          }
-       }
-    }
-  ]
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "NonResourceBasedReadOnlyPermissions",
```
Comment (mattbostock) [inline projects/packer_ami_builder/resources/production/templates/ami_building_policy.tpl]: Can these `Sid` be made more descriptive? For example, I think many of these actions are not read-only. It'd be good if the Sid could help us to distinguish between the groups of actions there are here.

**Comment by mattbostock** [comment]
@jonauman: We should probably scope f18b2e7 to just the subnet and VPC IDs for now, which are the variables will may change and are introduced in this pull request.

I also wonder if those variables should be null, so that they're mandatory and Terraform will fail fast if they're not supplied.

**Comment by mattbostock** [comment]
Generally we try to avoid merge commits in branches such as 2cf8499 because they lack any explanation for the changes in the code. Happy to pair to help remove it.

**Comment by mattbostock** [comment]
I think this is looking good now but I'll leave this for someone else with more context to review the EC2 permissions.

**Pull request merged**